packages/framework/src/DependencyInjection/Compiler/RegisterProjectFrameworkClassExtensionsCompilerPass.php

```diff
@@ -16,11 +20,26 @@ class RegisterProjectFrameworkClassExtensionsCompilerPass implements CompilerPas
     public function process(ContainerBuilder $container): void
     {
         $classExtensionRegistryDefinition = $container->findDefinition(ClassExtensionRegistry::class);
+
         foreach ($container->getServiceIds() as $serviceId) {
             if ($this->isFrameworkClassWithAlias($container, $serviceId)) {
                 $aliasId = (string)$container->getAlias($serviceId);
+
+                try {
+                    $serviceClass = $container->findDefinition($serviceId)->getClass();
+                } catch (ServiceNotFoundException) {
+                    $serviceClass = null;
+                }
+
+                if ($serviceClass !== null && !$this->isProjectClass($aliasId) && $this->isProjectClass($serviceClass)) {
```

Comment:
This check is done only if the service is already aliased... meaning this condition can be met only if
- current service is a framework class
- current service has the alias defined
- the alias is not a project class
- current service has the class set as a project class

how should look the definition to meet these conditions?

Reply:
This happens when class is extended in services.yaml instead of aliasing.

Shopsys\FrameworkBundle\Model\Customer\User\CustomerUserUpdateDataFactory:
    class: App\Model\Customer\User\CustomerUserUpdateDataFactory

Reply:
This class will work, because it implements the interface and this interface has defined alias in services.yaml (https://github.com/shopsys/shopsys/blob/master/packages/framework/src/Resources/config/services.yaml#L756) 

I tried it this way: 
- create new `\App\Model\Product\TopProduct\TopProductFacade` class extending the framework's one
- create new `\App\Controller\Admin\TopProductController`class extending the framework's one
- extend it in services.yaml with the `class`
    ```
    Shopsys\FrameworkBundle\Model\Product\TopProduct\TopProductFacade:
        class: App\Model\Product\TopProduct\TopProductFacade
    ```
- run `php phing annotations-fix`

In the current implementation, only `@method \App\Model\Product\Product[] getProductsForDomain(int $domainId)` annotation is added. 

Expected is also the annotation for the constructor and property `$topProductFacade`

Reply:
I have tested it on our internal project and I got the `You have requested a non-existent service "App\Model\Mail\MailTemplateConfiguration".` error.

Can you check it, please?